grafana/explore-profiles
# refactor(*): Improve error tracking

### ✨ Description

**Related issue(s):** resolves https://github.com/grafana/explore-profiles/issues/263

This PR is an attempt to improve how we track errors, in order to benefit from great frontend observability dashboards and fast response in case of incidents!

### 📖 Summary of the changes

Besides adding an `ErrorBoundary` with a new error page:

<img width="1707" alt="image" src="https://github.com/user-attachments/assets/9426668d-81c5-4d26-a907-7c177d6e9fd6">

This PR introduces several changes:
- A new `Logger` class, which encapsulates calls to Faro and/or `console.xyz()`
- All existing `console.xyz()` calls have been changed to calls to `logger.xyz()`
- Enforcing logger usage: ESLint will fail if any `console.xyz()` calls are present in the code
- When calling `displayError()`, the error is automatically logged
- When passing an `error` as prop to `<InlineBanner />`, it's automatically logged
- Refactor: `<InlineBanner />` now only accepts a single error as prop (no more arrays)

### 🧪 How to test?

The build should pass: some unit tests have been added, some refactored.

Manually:
- checkout this PR's branch locally,
- uncomment [these lines in faro-environments.ts](https://github.com/grafana/explore-profiles/pull/264/files#diff-920c144b32a862981a964657432fec328bff344602b837ee6678f30349ee0281R6-R13)
- manually force errors by modifying the code and playing with the app
- go to our "Frontend Observability" dashboard and see the errors and their the associated data (error context, ...)
